app/lib/frontend/templates/package.dart

```diff
@@ -292,12 +263,57 @@ String renderPkgShowPage(
   );
 }
 
-List<Map<String, String>> _tabsData(
+/// Renders the `views/pkg/tabs.mustache` template.
+String renderPkgTabs(List<Tab> tabs) {
+  final tabsData = tabs.map((t) => t.toMustacheData()).toList();
+  // active: the first one with content
+  for (int i = 0; i < tabs.length; i++) {
+    if (tabs[i].contentHtml != null) {
+      tabsData[i]['active'] = '-active';
+      break;
+    }
+  }
+  final values = {'tabs': tabsData};
+  return templateCache.renderTemplate('pkg/tabs', values);
+}
+
+/// Defines the header and content part of a tab.
+class Tab {
+  final String id;
+  final String titleHtml;
+  final String contentHtml;
+  final bool isMarkdown;
+
+  Tab.withContent({
+    @required this.id,
+    String title,
+    String titleHtml,
+    @required this.contentHtml,
+    this.isMarkdown = false,
+  }) : titleHtml = titleHtml ?? htmlEscape.convert(title);
+
+  Tab.withLink({
+    String title,
+    String titleHtml,
+    @required String href,
+  })  : titleHtml =
+            '<a href="$href">${titleHtml ?? htmlEscape.convert(title)}</a>',
```

Comment:
Are we properly escaping the `href` here?

Reply:
This is a url that we generate, we don't escape these in most places.